forms/models/elements/subform.js

```diff
@@ -256,7 +256,14 @@ define(function (require) {
 
       return new Promise(function (resolve, reject) {
         Promise.all(promises).then(function (values) {
-          resolve(values);
+          values = _.reduce(values, function (memo, value) {
+            if (value._action === 'remove' && (value.id === null || value.id === undefined)) {
```

Comment:
I think `null` || `undefined` is too specific for this test. It would be better to ensure that `value.id` is either a non-empty String or a Number.